### DISCUSSION TITLE
Theia backend ports to alternative JS runtimes?

### DISCUSSION THREAD

**Discussion by decalek** [eclipse-theia/theia#10284] [Q&A] [answered]
Are there any efforts/ideas to port Theia backend to alternative JS runtime (e.g. [es4x](https://github.com/reactiverse/es4x), Deno)?

**Comment by vince-fugnitto** [comment, marked as answer]
@decalek thank you for the discussion!

As far as I know there are no official plans or ongoing efforts to use or support alternative JS runtimes for the backend.
There are many factors to take into consideration, and the backend technology should work well for both the browser and desktop (electron) use-case. Given all the effort, I do not think it will be planned anytime soon.

**Reply by decalek** [reply]
Thank you, Vincent! I fully understand that the core team have much more important goals in the near future roadmap. IMHO, such eventual ports should be developed as separate, downstream projects by the respective interested communities - vert.x, deno, some quickjs-based, etc. If someone is interested - Please, place a note here!

The maximum that upstream project could provide is technical assistance and coordination, like:
 * Advices which parts of code and documentation should be learned first
 * Sharing the recipes/result of backend codebase analysis tools (if/when the project uses any): e.g. Index of the uses of Node.js C/JS APIs
 * Guidance regarding need of writing additional tests, etc.
